api/src/test/java/org/openmrs/module/patientqueueing/api/PatientQueueingServiceTest.java

```diff
@@ -22,140 +22,186 @@
+
+        Location location = Context.getLocationService().getLocation(1);
+
+        List<PatientQueue> originalPatientQueueList = patientQueueingService.getPatientQueueList(null, null, null, null, null, null, null);
+
+        PatientQueue patientQueue = new PatientQueue();
+        patientQueue.setPatient(patient);
+        patientQueue.setStatus(PatientQueue.Status.PENDING);
+        patientQueue.setVisitNumber("QN-001");
+        patientQueue.setEncounter(Context.getEncounterService().getEncounter(10000));
+        patientQueue.setLocationFrom(location);
+        patientQueue.setLocationTo(location);
+        patientQueue.setPriority(0);
+        patientQueue.setPriorityComment("Emergency");
+        patientQueueingService.savePatientQue(patientQueue);
+
+        List<PatientQueue> newPatientQueueList = patientQueueingService.getPatientQueueList(null, null, null, null, null, null, null);
+
+        Assert.assertEquals(originalPatientQueueList.size() + 1, newPatientQueueList.size());
+
+        Assert.assertEquals("QN-001", newPatientQueueList.get(0).getVisitNumber());
+
+    }
+
+    @Test
+    public void savePatientQueue_shouldCompleteExistingPatientQueueWithSamePatientAndLocation() throws Exception {
+
+        PatientQueueingService patientQueueingService = Context.getService(PatientQueueingService.class);
+
+        Patient patient = Context.getPatientService().getPatient(10000);
+
+        Location location = Context.getLocationService().getLocation(1);
+
+        List<PatientQueue> originalPatientQueueList = patientQueueingService.getPatientQueueList(null, null, null, null, null, null, null);
+
+        PatientQueue patientQueue = new PatientQueue();
+        patientQueue.setPatient(patient);
+        patientQueue.setStatus(PatientQueue.Status.PENDING);
+        patientQueue.setVisitNumber("QN-001");
+        patientQueue.setEncounter(Context.getEncounterService().getEncounter(10000));
+        patientQueue.setLocationFrom(location);
+        patientQueue.setLocationTo(location);
+        patientQueue.setPriority(0);
+        patientQueue.setPriorityComment("Emergency");
+        PatientQueue patientQueue1 = patientQueueingService.savePatientQue(patientQueue);
+
+        PatientQueue patientQueue2 = new PatientQueue();
+        patientQueue2.setPatient(patient);
+        patientQueue2.setStatus(PatientQueue.Status.PENDING);
+        patientQueue2.setVisitNumber("QN-002");
+        patientQueue2.setEncounter(Context.getEncounterService().getEncounter(10000));
+        patientQueue2.setLocationFrom(location);
+        patientQueue2.setLocationTo(location);
+        patientQueue2.setPriority(0);
+        patientQueue2.setPriorityComment("Emergency");
+        patientQueueingService.savePatientQue(patientQueue2);
+
+        Assert.assertEquals(PatientQueue.Status.COMPLETED, patientQueue1.getStatus());
+
+    }
+
+    @Test
+    public void generateVisitNumber_shouldReturnNewvisitNumberForGivenPatientOnGivenDate() {
```

Comment:
I find this test method name to be the opposite of the one below it, without a reason in the method name. Hence confusing.